_posts/2020-08-23-gsoc-20-tf-2-examples.md

```diff
@@ -0,0 +1,66 @@
+
+# The Kubeflow Community
+
+The [Kubeflow community](https://www.kubeflow.org/docs/about/community/) is a highly approachable and closely-knit community that has been reaching out to and [helping](https://www.kubeflow.org/docs/about/gsoc/) potential GSoC students well before the application period. Respecting this, I made sure to take feedback for my proposal of the [project idea](https://summerofcode.withgoogle.com/projects/#5507335985823744) I chose, before the application deadline. Mentors [Yuan Tang](https://github.com/terrytangyuan), [Ce Gao](https://github.com/gaocegege) and [Jack Lin](https://github.com/ChanYiLin) were candid in providing me feedback and I refined and changed my proposal accordingly. To my sweet surprise, I got selected for the idea!😁 What has really helped me in these three months of coding period is that one month of [community bonding](https://developers.google.com/open-source/gsoc/timeline) where I got to know the community and more about the technicalities of Kubeflow.
+
+# The Project
+
+![](/images/2020-08-23-gsoc-20-tf-2-examples/kubeflow_components.png "Kubeflow Components")
+
+Examples created as part of this project needed to be easily reproducible to serve their purpose. Initially the underlying model decided to demonstrate Kubeflow functionalities was a BiDirectional RNN to be trained on IMDB large movie review [dataset](http://ai.stanford.edu/%7Eamaas/data/sentiment/) for sentiment analysis based on a [tensorflow tutorial](https://www.tensorflow.org/tutorials/text/text_classification_rnn). Over the course of time, we decided to also add another set of examples using  a neural machine translation model in its backend trained on a Spanish to English [dataset](http://www.manythings.org/anki/) based on another [tensorflow tutorial](https://www.tensorflow.org/tutorials/text/nmt_with_attention).
+
+The reasons for choosing these models were:
+* The [kubeflow/examples](https://github.com/kubeflow/examples) repo needed more NLP-related tasks.
+* These were more of *hello world* tasks in the field of NLP. So that users who go through these samples need not worry about training code and focus more on Kubeflow's functionalities.
+* These are based on tensorflow tutorials. Kubeflow tutorials based on Tensorflow tutorials show better coupling between the two.
+
+## Repository Structure
+
+I created a [repo](https://github.com/yashjakhotiya/kubeflow-gsoc-2020) under my own profile to regularly push commits to and my mentors consistently reviewed the work I pushed there. This repo has all of my work with the log history preserved. Each of the two models has following notebooks explaining core Kubeflow functionalities - 
+
+1. `<training-model>.py` - This is the core training code upon which all subsequent examples showing Kubeflow functionalities are based. Please go through this first to know more about the machine learning task subsequent notebooks will manage.
```

Comment:
Maybe add links for each component to find it easier in repository for the reader? 
For example for `text_classification` model.

What do you think @yashjakhotiya ?

Reply:
Each component would exist for both models - `text_classification` and `neural_machine_translation` (and maybe more in the future). In that case, how do you propose to add links @andreyvelich?

Reply:
Maybe chose one model and add the links ?
Something like this:
"For example check the [source code](https://github.com/yashjakhotiya/kubeflow-gsoc-2020/blob/master/text_classification/distributed_text_classification_rnn.py) for `text_classification` model."

Reply:
Okay, I will add relevant links. Thanks for the review!